test/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/initialgainorloss/InitialGainOrLossControllerSpec.scala

```diff
@@ -345,9 +446,42 @@ class InitialGainOrLossControllerSpec
       }
 
       "have proper contents" in {
+        val draftReturn = sample[DraftSingleDisposalReturn]
+          .copy(initialGainOrLoss = Some(AmountInPence(1L)))
+
+        val fillingOutReturn      = sample[FillingOutReturn].copy(draftReturn = draftReturn)
+        val trustFillingOutReturn = fillingOutReturn.copy(subscribedDetails   = generateTrustSubscribedDetails())
+        val individualFillingOutReturn = fillingOutReturn.copy(
+          subscribedDetails = generateIndividualSubscribedDetails(),
+          draftReturn       = draftReturn.copy(triageAnswers = generateTriageAnswersWithSelf())
+        )
+        val personalRepresentativeFillingOutReturn = fillingOutReturn.copy(draftReturn =
+          draftReturn.copy(triageAnswers = generateTriageAnswersWithPersonalRepresentative())
+        )
+        val capacitorFillingOutReturn = individualFillingOutReturn.copy(draftReturn =
+          draftReturn.copy(triageAnswers = generateTriageAnswersWithCapacitor())
+        )
+
+        val testData = List(
+          ("", individualFillingOutReturn, false),
+          (".trust", trustFillingOutReturn, false),
+          (".personalRep", personalRepresentativeFillingOutReturn, false),
+          (".capacitor", capacitorFillingOutReturn, false),
+          (".agent", individualFillingOutReturn, true)
+        )
+        testData.map(keyWithJourneyStatus => test(keyWithJourneyStatus._1, keyWithJourneyStatus._2, keyWithJourneyStatus._3))
```

Comment:
it makes more sense to use `foreach` instead of `map` here and elsewhere in this spec where you're doing a similar thing - using `map` you end up with a `List[Unit]`, but we would ideally end up with just `Unit` since `List[Unit]` doesn't mean anything useful

Reply:
Done.